src/utils/paymentQrValidation.ts

```diff
@@ -43,15 +67,24 @@ const isValidPaymentQR = (payload: string): boolean => {
     Sentry.captureException(e);
 
     if (e instanceof PaymentQRUnsupportedError) {
-      throw new PaymentQRUnsupportedError(
-        ERROR_MESSAGE.INVALID_IDENTIFIER_INPUT
-      );
+      throw isTextInputDisabled
+        ? new PaymentQRUnsupportedErrorTextDisabled(
+            ERROR_MESSAGE.INVALID_IDENTIFIER_INPUT_TEXT_DISABLED
+          )
+        : new PaymentQRUnsupportedError(ERROR_MESSAGE.INVALID_IDENTIFIER_INPUT);
```

Comment:
for the 3 scenarios here, can we just throw the same error, but different error message? 

e.g. 
```suggestion
      throw isTextInputDisabled
        ? new PaymentQRUnsupportedError(
            ERROR_MESSAGE.INVALID_IDENTIFIER_INPUT_TEXT_DISABLED
          )
        : new PaymentQRUnsupportedError(ERROR_MESSAGE.INVALID_IDENTIFIER_INPUT);
```

Reply:
oops, just to clarify, what i meant was that we can keep the specific QR errors, (i.e. `PaymentQRUnsupportedError`, `PaymentQRMissingInfoError`), but we dont need the errors that include the `TextDisabled` (i.e. `PaymentQRUnsupportedErrorTextDisabled`)

Reply:
Okay! code updated.